File: SecondYear/semestrul4/MPP/PROIECT/JavaProjectORM/ProjectModel/src/main/java/ro/mpp2025/javaprojectui/dto/ProbaDTO.java
package ro.mpp2025.javaprojectui.dto;


import ro.mpp2025.javaprojectui.CategorieVarsta;
import ro.mpp2025.javaprojectui.Entity;
import ro.mpp2025.javaprojectui.Proba;

import java.io.Serializable;

public  class ProbaDTO extends Entity<Integer> implements Serializable {
    private final Proba proba;
    private final int registeredCount;

    public ProbaDTO(Proba proba, int registeredCount) {
        this.proba = proba;
        this.registeredCount = registeredCount;
    }

    public Proba getProba() {
        return proba;
    }

    public String getEventName() {
        return proba.getTip();
    }

    public String getAgeGroup() {
        CategorieVarsta cv = proba.getVarsta();
        return cv.getVarstaMin() + "-" + cv.getVarstaMax() + " ani";
    }

    public int getRegisteredCount() {
        return registeredCount;
    }

    @Override
    public String toString() {
        return "ProbaDTO{" +
                "proba=" + proba +
                ", registeredCount=" + registeredCount +
                '}';
    }
}

File: SecondYear/semestrul4/MPP/PROIECT/CSharpProjectNetworking/Model/dto/ProbaDTO.cs
namespace Model;
public class ProbaDTO
{
    public Proba Proba { get; set; }
    public int NumarInregistrati { get; set; }

    public ProbaDTO() {}
    public ProbaDTO(Proba proba, int numarInregistrati)
    {
        Proba = proba;
        NumarInregistrati = numarInregistrati;
    }
    

    public Proba GetProba()
    {
        return Proba;
    }

    public string GetNumeEveniment()
    {
        return Proba.Tip;
    }

    public string GetGrupaVarsta()
    {
        CategorieVarsta cv = Proba.Varsta;
        return cv.VarstaMin + "-" + cv.VarstaMax + " ani";
    }

    public int GetNumarInregistrati()
    {
        return NumarInregistrati;
    }
    
    public override string ToString()
    {
        return $"Proba: {Proba}, NumarInregistrati: {NumarInregistrati}";
    }
}